src/SWP/TemplateEngineBundle/Twig/Node/GimmeListNode.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace SWP\TemplateEngineBundle\Twig\Node;
+
+/**
+ * Gimme twig node.
+ */
+class GimmeListNode extends \Twig_Node
+{
+    protected static $count = 1;
+
+    protected $loop;
+
+    public function __construct(\Twig_Node $variable, \Twig_Node $collectionType, \Twig_Node_Expression_Filter $collectionFilters = null, \Twig_Node_Expression $parameters = null, \Twig_Node_Expression $ifExpression = null, \Twig_NodeInterface $else = null, \Twig_NodeInterface $body, $lineno, $tag = null)
+    {
+        $body = new \Twig_Node(array($body, $this->loop = new \Twig_Node_ForLoop($lineno, $tag)));
+
+        if (null !== $ifExpression) {
+            $body = new \Twig_Node_If(new \Twig_Node(array($ifExpression, $body)), null, $lineno, $tag);
+        }
+
+        parent::__construct(array(
+            'variable' => $variable, 
+            'collectionType' => $collectionType,
+            'collectionFilters' => $collectionFilters,
+            'parameters' => $parameters,
+            'ifExpression' => $ifExpression,
+            'else' => $else,
+            'body' => $body
+        ), array('with_loop' => true, 'ifexpr' => null !== $ifExpression), $lineno, $tag);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function compile(\Twig_Compiler $compiler)
+    {
+        $i = self::$count++;
```

Comment:
Then in compiler i would need write this: 
```
$compiler->write("\$swpCollectionMetaLoader".$compiler."
```
instead 
```
$compiler->write("\$swpCollectionMetaLoader".$i."
```
every time, It's not needed otpimization. http://programmers.stackexchange.com/a/86911